enhancements/20220614-sriov-configuration.md

```diff
@@ -0,0 +1,125 @@
+      resourceName: "nvidia.com/TU104GL_Tesla_T4"
+      externalResourceProvider: true
+    - pciVendorSelector: "8086:6F54"
+      resourceName: "intel.com/qat"
+    mediatedDevices:
+    - mdevNameSelector: "GRID T4-1Q"
+      resourceName: "nvidia.com/GRID_T4-1Q"
+```
+
+Then the KubeVirt VM in your harvester cluster will have direct access to the underlying PCI device, and it will be possible to enable SR-IOV in the guest cluster.
+
+TODO: detailed explanation of how to do this
+
+
+## Technical details
+Steps to enable PCI Passthrough mode, which gives direct access to your PCI device, 
+avoiding the hypervisor translation layer.
+
+1. Enable IOMMU extensions in your BIOS
+2. Enable the iommu and iommu.passthrough kernel parameters 
+3. Unbind the driver from the PCI device 
+4. Bind the `vfio-pci` driver to the PCI device
+5. Modify the KubeVirt CR to include the permitted mdevs (mediated devices)
+
+### 
+
+# Host cluster mode
+    
+## User Experience
+
+The Harvester cluster admin starts by installing Harvester. By default, the SR-IOV feature is disabled.
+
+If the user wants to enable the feature, they need to set the `sriov-networking-enabled` setting to `true`
+
+Admin wants to create an SR-IOV Network. To do so, they navigate to the Harvester Dashboard UI. 
+Admin expands the Advanced menu on the left side and looks right below "Networks", to see the "SR-IOV"
+tab. This will show the form to create the SriovNetworkNodePolicy and the SriovNetwork.
+
+After the node policy is created, the networks' `spec.resourceName` is set to the policy's `spec.resource`.
+
+At this point, admin can attach VMs to the SriovNetwork and begin taking advantage of the direct-memory access I/O provided by their NIC.
+
+## Requirements
+1. SR-IOV nav item (under Advanced) is only visible when `sriov-networking-enabled` setting to `true`
+2. Clicking on the SR-IOV nav item will show a form to edit the fields for these two CRDs:
+   2a. [SriovNetwork](https://docs.openshift.com/container-platform/4.6/networking/hardware_networks/configuring-sriov-net-attach.html) and
+   2b. [SriovNetworkNodePolicy](https://docs.openshift.com/container-platform/4.6/networking/hardware_networks/configuring-sriov-device.html**
+3. When the SriovNetworkNodePolicy is created, the spec.resourceName should be the default value for the SriovNetwork's spec.resourceName
+4. the VLAN number in the SriovNetwork must be unique from all the other VLAN numbers currently
+5. nodes UI needs checkbox for 'feature.node.kubernetes.io/network-sriov.capable=true' label
+
+## Technical Details
+To make use of the operator, we have a fork of the operator source code here. 
+Repo: https://github.com/harvester/sriov-network-operator
+
+### Installation
+When the `sriov-network-enabled` setting is changed to `true`, the settings controller will
+install the `sriov-network-operator` [Helm chart](https://github.com/harvester/sriov-network-operator/tree/master/deployment/sriov-network-operator).
```

Comment:
We may need a little bit more comparison between using the `sriov-network-operator` vs `vanilla sriov chart` like https://github.com/rancher/charts/pull/1211#pullrequestreview-668492770